Vehicle 삭제 API 구현

Shin-Dong-Jun opened the pull request:
### 📌 PR 타입(하나 이상의 PR 타입을 선택해주세요)
- 기능 추가 ✅
- 기능 삭제
- 버그 수정
- 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 📌 관련 이슈

### ✨ 반영 브랜치
feat/19 -> develop

### 📝 변경 사항
- [ ✅ ] 차량 삭제 기능 추가

### ➕ 추가 메모
- 차량 삭제시 이력을 남겨야하지 않을까 싶다.

### 🐛 테스트 결과 (테스트 결과가 있다면 넣어주세요)
[ControllerTest]
![image](https://github.com/user-attachments/assets/d849b2dc-db56-41f1-a3e1-30c068b13950)
![image](https://github.com/user-attachments/assets/0c5e2908-959b-495f-bc6b-baabf7b57582)
![image](https://github.com/user-attachments/assets/6ae96e9e-8753-4b53-8f28-473c6a029deb)

[ServiceTest]
![image](https://github.com/user-attachments/assets/373fd0b0-0d8d-4c11-b736-57a9c1b5fbb5)

Jeongyounghyeon committed:
feat: #16 Member-Company 연관관계 설정 및 등록 로직 수정

- Member 엔티티에 Company 연관관계(`@ManyToOne`) 설정
- Company 정보를 Member 조회 응답 DTO에 포함
- Member 등록 시 Company ID를 함께 받도록 요청 구조 및 생성 로직 수정

Jeongyounghyeon committed:
chore: #16 인증 기능 반영을 위한 Member 조회 todo 주석 추가

SeoYoonP committed:
feat: #31 로그데이터 생성 및 수정 서비스 구현

SeoYoonP committed:
feat: #31 jdbc Config 정의

SeoYoonP committed:
feat: #31 Log 생성 및 저장 단위 테스트

SeoYoonP committed:
feat: #31 Log 생성 스케줄러 서비스 정의

SeoYoonP committed:
refactor: #31 스케줄링 어노테이션 추가

Shin-Dong-Jun committed:
feat: #12 차량 상세 조회 API 구현

Shin-Dong-Jun committed:
fix: #19 rebase 중 충돌 해결

SeoYoonP committed:
refactor: #31 로그 생성 서비스 및 테스트코드 삭제

SeoYoonP committed:
feat: #31 로그 데이터 배치 요청을 위한 DTO 정의

SeoYoonP committed:
feat: #31 데이터 수신을 위한 로그 컨트롤러 구현

SeoYoonP committed:
feat: #31 수신 로그 데이터 저장 서비스 구현

SeoYoonP committed:
refactor: #31 @EnableScheduling 어노테이션 제거

SeoYoonP committed:
feat: #31 LogDataSaveServiceTest 구현

- 배치 로그 데이터 저장 성공
- 예외 발생시 로그 데이터 롤백

Shin-Dong-Jun committed:
feat: #33 Vehicle 등록 API(2차 리뷰 반영)

- 필드명 변경 isOn -> powerOn
- VehicleService의 registerVehicle 메서드에 Company 삭제여부 확인 추가

Shin-Dong-Jun committed:
feat: #19 차량 삭제 API 구현

Shin-Dong-Jun committed:
feat: #18 VehicleResponse 응답값 추가(companyId)

Jeongyounghyeon commented on src/main/java/org/thisway/vehicle/service/VehicleService.java:
```diff
@@ -36,4 +36,18 @@ public void registerVehicle(VehicleCreateRequest request) {
+
+        if (!vehicle.isActive()) {
+            throw new CustomException(ErrorCode.VEHICLE_ALREADY_DELETED);
+        }
```
이미 삭제된 데이터에 대해서 이미 삭제된 데이터와 없는 데이터를 구분해서 예외 처리할 것인지 정해봐야할 것 같습니다!

roomie00 replied:
이 부분 논의가 필요할 것 같아서 Approve는 우선 하지 않도록 하겠습니다 :)

Shin-Dong-Jun replied:
넵 확인했습니다!!

roomie00 replied:
PR#47 논의하며 리팩토링으로 넘기면 될 것 같습니다 :)

Jeongyounghyeon reviewed (approved):
👍

Jeongyounghyeon committed:
refactor: #42 `CompanyRegisterRequest` 사용되지 않는 `from` 메서드 삭제

Jeongyounghyeon committed:
refactor: #42 Company 삭제시 엔티티 활성 상태 확인 로직 추가

Jeongyounghyeon committed:
refactor: #42 의미없는 애너테이션 삭제

Jeongyounghyeon committed:
refactor: #42 `CompanyRegisterRequest` 클래스 엔티티로의 변환 메서드 생성

서비스 코드 가독성 향상을 위해 변환 메서드 생성

SeoYoonP commented on src/main/java/org/thisway/vehicle/controller/VehicleController.java:
```diff
@@ -23,4 +20,11 @@ public ApiResponse<Void> registerVehicle(@RequestBody @Validated VehicleCreateRe
+
+    @DeleteMapping("/{id}")
+    public ApiResponse<Void> deleteVehicle(@PathVariable Long id){
+
```
사소한것 하나 질문드리고 싶습니다. 클래스 명 아래에는 한 줄 비워놓기로 한 것 같은데 메소드 아래에 한 줄 비워놓으시는 이유는 가독성 때문이신가요? (작성해주신 모든 메서드에 해당되는 리뷰입니다!)

Shin-Dong-Jun replied:
넵 이건 저의 개인적인 성향입니다! 
그런데 띄워놓지 않으면 너~무 불편하다는 또 아니여서 전체적인 코드 스타일을 맞추기 위해서 한 줄 비워놓는것은 없애도록 하겠습니다!

SeoYoonP committed:
refactor: #31 규격서에 따른 구조 변경으로 인한 기존 Log클래스들 삭제

- LogBatchRequest.java
- LogDataEntry.java
- LogDataSaveService.java
- LogDataSaveServiceTest.java

SeoYoonP committed:
feat: #31 GpsLog 데이터 전달 객체 정의

- GpsLogRequest
- GpsLogEntry

SeoYoonP committed:
feat: #31 PowerLog 데이터 전달 객체 정의

SeoYoonP committed:
feat: #31 GeofenceLog 데이터 전달 객체 정의

SeoYoonP committed:
feat: #31 로그 데이터 응답 객체 정의

SeoYoonP committed:
feat: #31 Emulator 엔티티 정의

SeoYoonP committed:
feat: #31 EmulatorRepository 정의

SeoYoonP committed:
fix: rebase 중 충돌 해결

SeoYoonP committed:
feat: #31 데이터 유형에 따른 LogController api 구현

SeoYoonP committed:
feat: #31 데이터 저장하는 LogService 구현

Jeongyounghyeon committed:
refactor: #42 활성화 상태 DB 레벨에서 필터링하도록 변경

SeoYoonP committed:
feat: #31 LogService 단위 테스트 코드 작성

Shin-Dong-Jun committed:
fix: #19 rebase 중 VehicleResponse 응답 값 변경에 따른 테스트 코드 수정

Shin-Dong-Jun committed:
fix: #19 에러코드 변경으로 인한 테스트 코드 수정

Shin-Dong-Jun committed:
Merge branch 'develop' into feat/19
